compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/GPUTileSwizzleUtils.cpp

```diff
@@ -114,8 +128,18 @@ static TileSwizzle getIntrinsicSwizzleBeforeMovingCrossThreadOutermost(
   }
   // Next come `layout.thread` dims.
   for (auto [i, t] : llvm::enumerate(layout.thread)) {
-    if (t != 1) {
-      expand(swizzle, i, {Kind::CrossThread, t});
+    // If the thread has a stride of 0, then we need a dimension for it in the
+    // swizzle so we can distribute by more than a factor of 1 along the dim.
+    if (t != 1 || layout.tstrides[i] == 0) {
+      TileSwizzle::Dim tDim(Kind::CrossThread, t);
+      if (layout.tstrides[i] == 0) {
+        int16_t subgroupSize = getIntrinsicSubgroupSize(intrinsic);
```

Comment:
```suggestion
        int64_t subgroupSize = getIntrinsicSubgroupSize(intrinsic);
```
I think the return type is actually `int64_t`?